src/include/range.h

```diff
@@ -51,6 +49,22 @@ enum range_step_type {
 	ENABLE_SUBRANGE_STEPPING
 };
 
+typedef struct range range;
+
+struct range
+{
+	int start;
+	int end;
+	int step;
+	int count;
+	range *next;
+};
+
+/* Error message when we fail to allocate memory */
+#define MEM_ERR_MSG "Unable to allocate memory (malloc error)"
```

Comment:
This is the generic malloc error that is used throughout the scheduler.  I thought it would be better this way than a one-off error message for each malloc() failure.  It is used in a log_err() whenever we do a malloc().  We can either merge it into the range log_err calls themselves like most of the server/lib code, or we can move the message to some place more generic so it can be used elsewhere in PBS.  I like being consistent and generic myself.,

Reply:
I kept this text string as it is now. Let us decide on this, I will implement it accordingly.